app/models/mixins/ownership_mixin.rb

```diff
@@ -2,13 +2,16 @@ module OwnershipMixin
   extend ActiveSupport::Concern
 
   included do
-    belongs_to :evm_owner, :class_name => "User"
+    belongs_to :evm_owner, :class_name => "User", :foreign_key => :evm_owner_id
     belongs_to :miq_group
 
-    virtual_column :evm_owner_email,                      :type => :string,     :uses => :evm_owner
-    virtual_column :evm_owner_name,                       :type => :string,     :uses => :evm_owner
-    virtual_column :evm_owner_userid,                     :type => :string,     :uses => :evm_owner
-    virtual_column :owned_by_current_user,                :type => :boolean,    :uses => :evm_owner_userid
+    virtual_delegate :email, :name, :userid, :to => :evm_owner, :prefix => true, :allow_nil => true
+    virtual_attribute :owned_by_current_user, :boolean, :uses => :evm_owner, :arel => (lambda do |t|
+      #t.grouping(arel_attribute(:evm_owner_id)]).eq(User.current_user.try(:id)))
+      t.grouping(
+        Arel::Nodes::NamedFunction.new("LOWER", [arel_attribute(:evm_owner_userid)]).eq(User.current_userid.downcase)
+      )
+    end)
```

Comment:
This would also allow us to avoid having to change the `belongs_to`/`ar_virtual.rb` until we can get it to work flawlessly.  May or may not be a perk.